### DISCUSSION TITLE
Idea for better explanation about bundle size impact

### DISCUSSION THREAD

**Discussion by kimgiutae** [dcastil/tailwind-merge#497] [General]
Hello. As the docs explains in "When and how to use it":

"Increases bundle size: tailwind-merge relies on a large config (~5 kB out of the ~7 kB minified and gzipped bundle size) to understand which classes are conflicting. This might be limiting if you have tight bundle size constraints." (https://github.com/dcastil/tailwind-merge/blob/v2.5.5/docs/when-and-how-to-use-it.md#increases-bundle-size)

But I don't think this explanation is correct, at least incomplete.

As the way react frameworks works now (as recommended). The client bundle size may increase ONLY if calling the lib in a CLIENT component. Using in the default RSC should not affect performance at all.

I'm wrong?

**Comment by dcastil** [comment]
Hey @kimgiutae! 👋

You're right that this only matters when tailwind-merge ends up in the client bundle. I don't want to assume too much about the context in which tailwind-merge is used though. E.g. if someone is new to programming and just uses tailwind-merge in a vanilla JS app, talking about RSC in the tailwind-merge docs would be confusing. I think it's better to rely on devs to learn about client/server bundles in the docs of the framework that deals with these bundles, like in the Next.js docs.

**Reply by gregg-cbs** [reply]
How does this work in sveltekit?

**Reply by dcastil** [reply]
@gregg-cbs I'm not very familiar with SvelteKit unfortunately. I think the Svelte community can help you out better with SvelteKit-related questions. Here is their Discord: https://discord.com/invite/svelte

tailwind-merge is just a normal JS function which takes some strings as input and produces a string as output, like any other function you'd write yourself. So it will always work the same as the rest of the code you write.
